### ISSUE TITLE
Training ABINET on Custom Dataset

### ISSUE THREAD

**Issue by bely66** [open-mmlab/mmocr#1166]
Hi Everyone,
I'm using my own custom dataset to train ABINET on a non-latin language.
I went through the following steps:
Generated labels.txt for the data with the OCRDataset format.
Added a config file for the new dataset in configs/_base_/rec_datasets
Changed the dataset path to the new config in abinet_academic.py in configs/rec/
then added dict_list with the new language characters in the  configs/_base_/rec_models/abinet.py

I faced a problem because the num_chars is hardcoded in the file:
<img width="170" alt="image" src="https://user-images.githubusercontent.com/29528270/179532154-533c24f4-068d-4ce6-b113-a7183e18bd16.png">

I changed it to the following:
<img width="284" alt="image" src="https://user-images.githubusercontent.com/29528270/179532264-18de8b18-50da-42d3-9eda-325ea7a7f850.png">

I added len(dict_list)+1 because it was failing without adding one.

**I have Two Questions:**
1. Did I miss any steps that could make the training go wrong or not progress?
2. I'm Training using a one V100 GPU and I tested with a small dataset size of 10k samples and It'll take 1 day to train for 20 epochs, I'm sensing that's a very long time?

**Comment by gaotongxiao** [comment]
1. The steps are fine. Currently `num_chars` has to be hardcoded, but this issue will be eliminated in our upcoming release.
2. ABINet is huge and it took us 5 days to finish up the training on 2xA100 GPUs.

**Comment by bely66** [comment]
so the num_chars should be the exact number? because I have to add 1 for the training to run

**Comment by gaotongxiao** [comment]
@bely66 That's right. The extra character comes from the special "<BOS/EOS>" token.
